File: Project/ebweb2019/src/main/webapp/app/shared/model/pp-pay-vendor-bill.ts
import { Moment } from 'moment';

export interface IPPPayVendorBill {
    check?: boolean;
    referenceID?: string;
    typeID?: number;
    date?: Moment;
    dueDate?: Moment;
    noFBook?: string;
    noMBook?: string;
    invoiceNo?: string;
    currencyID?: string;
    totalDebitOriginal?: number;
    totalDebit?: number;
    debitAmountOriginal?: number;
    debitAmount?: number;
    account?: string;
    employeeName?: string;
    paymentClause?: string;
    refVoucherExchangeRate?: number;
    lastExchangeRate?: number;
    differAmount?: number;
    amount?: number;
    amountOriginal?: number;
    discountAmount?: number;
    discountAmountOriginal?: number;
    discountRate?: number;
    discountAccount?: string;
    amountTT?: number;
}

export class PPPayVendorBill implements IPPPayVendorBill {
    constructor(
        public check?: boolean,
        public referenceID?: string,
        public typeID?: number,
        public date?: Moment,
        public dueDate?: Moment,
        public noFBook?: string,
        public noMBook?: string,
        public invoiceNo?: string,
        public currencyID?: string,
        public totalDebitOriginal?: number,
        public totalDebit?: number,
        public debitAmountOriginal?: number,
        public debitAmount?: number,
        public account?: string,
        public paymentClause?: string,
        public refVoucherExchangeRate?: number,
        public lastExchangeRate?: number,
        public differAmount?: number,
        public amount?: number,
        public amountOriginal?: number,
        public discountAmount?: number,
        public discountAmountOriginal?: number,
        public discountRate?: number,
        public discountAccount?: string,
        public amountTT?: number
    ) {
        this.check = this.check || false;
    }
}

File: Project/ebweb2019/src/main/java/vn/softdreams/ebweb/domain/PPPayVendorBill.java
package vn.softdreams.ebweb.domain;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public class PPPayVendorBill {

    private UUID referenceID;

    private String date;

    private String noFBook;

    private String noMBook;

    private String invoiceNo;

    private String dueDate;

    private BigDecimal totalDebitOriginal;

    private BigDecimal totalDebit;

    private BigDecimal debitAmountOriginal;

    private BigDecimal debitAmount;

    private String account;

    private String currencyID;

    private Integer typeID;

    private String employeeName;

    private String paymentClause;

    private BigDecimal refVoucherExchangeRate;

    private BigDecimal lastExchangeRate;

    private BigDecimal differAmount;

    public UUID getReferenceID() {
        return referenceID;
    }

    public void setReferenceID(UUID referenceID) {
        this.referenceID = referenceID;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getNoFBook() {
        return noFBook;
    }

    public void setNoFBook(String noFBook) {
        this.noFBook = noFBook;
    }

    public String getNoMBook() {
        return noMBook;
    }

    public void setNoMBook(String noMBook) {
        this.noMBook = noMBook;
    }

    public String getInvoiceNo() {
        return invoiceNo;
    }

    public void setInvoiceNo(String invoiceNo) {
        this.invoiceNo = invoiceNo;
    }

    public String getDueDate() {
        return dueDate;
    }

    public void setDueDate(String dueDate) {
        this.dueDate = dueDate;
    }

    public BigDecimal getTotalDebitOriginal() {
        return totalDebitOriginal;
    }

    public void setTotalDebitOriginal(BigDecimal totalDebitOriginal) {
        this.totalDebitOriginal = totalDebitOriginal;
    }

    public BigDecimal getTotalDebit() {
        return totalDebit;
    }

    public void setTotalDebit(BigDecimal totalDebit) {
        this.totalDebit = totalDebit;
    }

    public BigDecimal getDebitAmountOriginal() {
        return debitAmountOriginal;
    }

    public void setDebitAmountOriginal(BigDecimal debitAmountOriginal) {
        this.debitAmountOriginal = debitAmountOriginal;
    }

    public BigDecimal getDebitAmount() {
        return debitAmount;
    }

    public void setDebitAmount(BigDecimal debitAmount) {
        this.debitAmount = debitAmount;
    }

    public String getAccount() {
        return account;
    }

    public void setAccount(String account) {
        this.account = account;
    }

    public String getCurrencyID() {
        return currencyID;
    }

    public void setCurrencyID(String currencyID) {
        this.currencyID = currencyID;
    }

    public Integer getTypeID() {
        return typeID;
    }

    public void setTypeID(Integer typeID) {
        this.typeID = typeID;
    }

    public String getEmployeeName() {
        return employeeName;
    }

    public void setEmployeeName(String employeeName) {
        this.employeeName = employeeName;
    }

    public String getPaymentClause() {
        return paymentClause;
    }

    public void setPaymentClause(String paymentClause) {
        this.paymentClause = paymentClause;
    }

    public BigDecimal getRefVoucherExchangeRate() {
        return refVoucherExchangeRate;
    }

    public void setRefVoucherExchangeRate(BigDecimal refVoucherExchangeRate) {
        this.refVoucherExchangeRate = refVoucherExchangeRate;
    }

    public BigDecimal getLastExchangeRate() {
        return lastExchangeRate;
    }

    public void setLastExchangeRate(BigDecimal lastExchangeRate) {
        this.lastExchangeRate = lastExchangeRate;
    }

    public BigDecimal getDifferAmount() {
        return differAmount;
    }

    public void setDifferAmount(BigDecimal differAmount) {
        this.differAmount = differAmount;
    }
}
